### ISSUE TITLE
Saving a draft reports success even if it fails

### ISSUE THREAD

**Issue by st3iny** [nextcloud/mail#6157]
@GretaD @st3iny one tiny thing I noticed is that even if saving the outbox revision (draft) fails, e.g. because of a 404 with the stubs in here, the UI still says *Draft saved*. Could you have a look or open a ticket?

_Originally posted by @ChristophWurst in https://github.com/nextcloud/mail/issues/5970#issuecomment-1077459585_

**Comment by nimishavijay** [comment]
Good point, if the draft is not saved we shouldn't show a "Draft saved" message.
Instead if draft is not saved:
- add a "Save draft" button to the left of "Discard draft"
- show message "Draft could not be saved"
- on clicking "Save draft" if there are any issues with saving then we could show the message "Draft could not be saved. Please try again".

Additionally we should move the message to the left so that the positions do not change. Here are some mockups for reference.

![image](https://user-images.githubusercontent.com/52440189/161061962-8c7a9f12-25d9-44d4-9dc8-60d95787e31a.png)
![image](https://user-images.githubusercontent.com/52440189/161062043-cce10f99-c4cb-48ce-b3fe-f32c1727d60a.png)

cc @jancborchardt
